storage/netapp/netapp_e_storagepool.py

```diff
@@ -0,0 +1,888 @@
+        # map the storage pool name to its id
+        try:
+            (rc, resp) = request(self.api_url + "/storage-systems/%s/storage-pools" % (self.ssid),
+                                 headers=dict(Accept="application/json"), url_username=self.api_usr,
+                                 url_password=self.api_pwd, validate_certs=self.validate_certs)
+        except Exception:
+            err = get_exception()
+            rc = err.args[0]
+            if rc == 404 and self.state == 'absent':
+                self.module.exit_json(
+                    msg="Storage pool [%s] did not exist." % (self.name))
+            else:
+                err = get_exception()
+                self.module.exit_json(
+                    msg="Failed to get storage pools. Array id [%s].  Error[%s]. State[%s]. RC[{3}]." % (
+                        self.ssid, str(err), self.state, rc))
+
+        self.debug("searching for storage pool '%s'" % storage_pool_name)
+
+        pool_detail = next(select(lambda a: a['name'] == storage_pool_name, resp), None)
+
+        if pool_detail:
+            found = 'found'
+        else:
+            found = 'not found'
+        self.debug(found)
+
+        return pool_detail
+
+    def get_candidate_disks(self):
+        self.debug("getting candidate disks...")
+
+        # driveCapacityMin is broken on /drives POST. Per NetApp request we built our own
+        # switch back to commented code below if it gets fixed
+        # drives_req = dict(
+        #     driveCount = self.criteria_drive_count,
+        #     sizeUnit = 'mb',
+        #     raidLevel = self.raid_level
+        # )
+        #
+        # if self.criteria_drive_type:
+        #     drives_req['driveType'] = self.criteria_drive_type
+        # if self.criteria_disk_min_aggregate_size_mb:
+        #     drives_req['targetUsableCapacity'] = self.criteria_disk_min_aggregate_size_mb
+        #
+        # # TODO: this arg appears to be ignored, uncomment if it isn't
+        # #if self.criteria_disk_min_size_gb:
+        # #    drives_req['driveCapacityMin'] = self.criteria_disk_min_size_gb * 1024
+        # (rc,drives_resp) = request(self.api_url + "/storage-systems/%s/drives" % (self.ssid), data=json.dumps(drives_req), headers=self.post_headers, method='POST', url_username=self.api_usr, url_password=self.api_pwd, validate_certs=self.validate_certs)
+        #
+        # if rc == 204:
+        #     self.module.fail_json(msg='Cannot find disks to match requested criteria for storage pool')
+
+        # disk_ids = [d['id'] for d in drives_resp]
+
+        try:
+            (rc, drives_resp) = request(self.api_url + "/storage-systems/%s/drives" % (self.ssid), method='GET',
+                                        url_username=self.api_usr, url_password=self.api_pwd,
+                                        validate_certs=self.validate_certs)
+        except:
+            err = get_exception()
+            pool_id = self.pool_detail['id']
```

Comment:
Does this need to be set in __init__?  Can't tell if this could be accessed before being assigned, but could potentially be a problem.